### PR TITLE
Lpd 33604 Retrieve document metadata sets via headless API

### PR DESCRIPTION
**Pull request by SamZiemer**
# What is this trying to solve?
https://liferay.atlassian.net/browse/LPD-33604

I am adding new headless entities and get methods for Document Metadata Sets.

# How am I fixing it?
Create a new entity and flesh out the full get methods for Document Metadata Sets

# How can you verify that it works?
Run the tests in `DocumentMetadataSetResourceTest`

# Additional Notes
I resubmitted this PR from https://github.com/liferay-content-management/liferay-portal/pull/5776 because I force pushed a few times and I wanted to make sure this final PR was clean without multiple test failures and force pushes.

### CODE REVIEW

**Comment by SamZiemer** [comment]
ci:test:relevant

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/DocumentMetadataSetDTOConverter.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+				setId(ddmStructure::getStructureId);
+
+				setName(
+					() -> ddmStructure.getName(
+						dtoConverterContext.getLocale()));
+
+				setName_i18n(
+					() -> LocalizedMapUtil.getI18nMap(
+						ddmStructure.getNameMap()));
+				setSiteId(ddmStructure::getGroupId);
```
Comment (adolfopa) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/DocumentMetadataSetDTOConverter.java:92]: Not saying it is wrong, but those blank lines between statements look weird. Did the SF ask you do that?

**Comment by adolfopa** [comment]
ci:test:stable

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/DocumentMetadataSetDTOConverter.java (hunk 1 above)
Comment (SamZiemer) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/DocumentMetadataSetDTOConverter.java]: I think you're right, it does look a bit odd, I've added a commit removing the extra lines. SF did not complain about the extra lines, but also does not complain when they are not there.

File: modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/DocumentMetadataSetResourceTest.java [hunk 2]
```diff
@@ -6,15 +6,240 @@
...
+						ddmStructure.getDescriptionMap()));
+				setId(ddmStructure.getStructureId());
+
+				setName(
+					() -> ddmStructure.getName(
+						testGroup.getDefaultLanguageId()));
+
+				setName_i18n(
+					() -> LocalizedMapUtil.getI18nMap(
+						ddmStructure.getNameMap()));
```
[the remark this answers is not in the document]
Comment (SamZiemer) [reply modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/DocumentMetadataSetResourceTest.java]: Same comment above, added a new commit to change this.

**Comment by SamZiemer** [comment]
ci:test:sf

**Comment by SamZiemer** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by SamZiemer** [comment]
ci:test:sf

**Comment by SamZiemer** [comment]
ci:test:relevant

**Comment by SamZiemer** [comment]
Thank you @adolfopa, I've fixed the issues with the tests and manually testing works as well. I believe this is good to go.

**Comment by SamZiemer** [comment]
The test failures look like they were caused from changes in upgrade. I've rebased and will rerun the tests.

**Comment by SamZiemer** [comment]
ci:test:relevant

**Comment by SamZiemer** [comment]
ci:test:sf

File: modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/DocumentMetadataSetResourceTest.java [hunk 3]
```diff
@@ -124,22 +125,11 @@ private DocumentMetadataSet _addDocumentMetadataSet(Group group)
...
-							dataDefinitionResource.getDataDefinition(
-								ddmStructure.getStructureId());
-
-						return dataDefinition.getDataDefinitionFields();
-					});
+					() -> DataDefinitionFieldSerDes.toDTOs(
+						StringBundler.concat(
+							StringPool.OPEN_BRACKET,
+							ddmStructure.getDefinition(),
+							StringPool.CLOSE_BRACKET)));
```
Comment (adolfopa) [inline modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/DocumentMetadataSetResourceTest.java:132]: Not sure about this. If you look at com.liferay.data.engine.rest.internal.resource.v2_0.DataDefinitionResourceImpl#_normalizeDataDefinitionFields, there's some additional processing they do to the ddm structure when converting to data engine fields. Can you double check it is correct?

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Comment by SamZiemer** [comment]
ci:test:sf

**Comment by SamZiemer** [comment]
ci:test:relevant

**Comment by SamZiemer** [comment]
ci:test:sf

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by adolfopa** [comment]
ci:report:80379423

**Comment by adolfopa** [comment]
ci:reevaluate:1307401_10046

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by SamZiemer** [comment]
ci:test:relevant

**Comment by SamZiemer** [comment]
ci:test:sf

**Comment by SamZiemer** [comment]
ci:test:sf

**Comment by SamZiemer** [comment]
ci:test:sf

**Comment by SamZiemer** [comment]
ci:test:relevant

**Comment by SamZiemer** [comment]
ci:reevaluate:1377401_17852

**Comment by SamZiemer** [comment]
ci:test:relevant

**Comment by SamZiemer** [comment]
The test failures do no look related, and the tests previously passed before a minor schema version change was required. I am going to force forward this pull.

**Comment by SamZiemer** [comment]
ci:forward:force

**Pull request closed without merging**